### ISSUE TITLE
IDE0007 reported for stackalloc to Span<T>

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#22768]
**Version Used**: 15.5 Preview 1

**Steps to Reproduce**:

```csharp
Span<int> span = stackalloc int[3];
```

**Expected Behavior**:

IDE0007 is not reported, because a change to use `var` also changes the type of `span`.

**Actual Behavior**:

IDE0007 is reported, and the code fix causes the code to break.

**Comment by jcouv** [comment]
FYI @OmarTawfik

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/22411 (warn for use of `stackalloc` in `var` local).

**Comment by sharwell** [comment]
It's similar, but this one appears to be a bug in the diagnostic analyzer not the compiler.

**Comment by jcouv** [comment]
#22411 would be primarily an analyzer and codefix too. We might add a compiler warning later on top of that.

**Comment by OmarTawfik** [comment]
@jcouv are you planning to work on this one? I can take it otherwise.

**Comment by jcouv** [comment]
Go for it!

**Comment by sharwell** [comment]
@jinujoseph Should we bring this into 15.5 since it impacts usability of one of the key new features of the release?
